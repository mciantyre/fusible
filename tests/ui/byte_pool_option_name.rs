// SPDX-License-Identifier: MPL-2.0
// SPDX-FileCopyrightText: Copyright 2024 Ian McIntyre

use fusible::byte_pool::{AlignedBytePoolStorage, BytePool, BytePoolOptions};
use std::ffi::CString;
use std::pin::pin;

fn main() {
    let mut storage = AlignedBytePoolStorage::from_array([0u8; 512]);
    let pool = pin!(BytePool::context());

    let name = CString::new("name").unwrap();
    let mut opts = BytePoolOptions::default();
    opts.name = Some(&name);

    BytePool::create(pool.into_ref(), &mut storage, &opts).unwrap();
}
