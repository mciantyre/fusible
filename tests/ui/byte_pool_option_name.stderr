error[E0597]: `name` does not live long enough
  --> tests/ui/byte_pool_option_name.rs:14:22
   |
10 |     let pool = pin!(BytePool::context());
   |                ------------------------- a temporary with access to the borrow is created here ...
11 |
12 |     let name = CString::new("name").unwrap();
   |         ---- binding `name` declared here
13 |     let mut opts = BytePoolOptions::default();
14 |     opts.name = Some(&name);
   |                      ^^^^^ borrowed value does not live long enough
...
17 | }
   | -
   | |
   | `name` dropped here while still borrowed
   | ... and the borrow might be used here, when that temporary is dropped and runs the `Drop` code for type `BytePoolContext`
